File: BE-FindingJob/src/main/java/com/example/befindingjob/service/NotificationService.java
package com.example.befindingjob.service;

import com.example.befindingjob.dto.NotificationDTO;
import com.example.befindingjob.dto.NotificationEntityDTO;
import com.example.befindingjob.entity.Notification;
import com.example.befindingjob.model.ApiResponse;

import java.util.List;

public interface NotificationService {
    ApiResponse<Void> createNotification(NotificationDTO notificationDTO);
    ApiResponse<List<NotificationEntityDTO>> getNotification(String token);
}

File: JobHub/app/src/main/java/com/example/jobhub/service/NotificationService.kt
package com.example.jobhub.service

import com.example.jobhub.dto.MarkAsReadDTO
import com.example.jobhub.dto.NotificationDTO
import com.example.jobhub.dto.NotificationEntityDTO
import com.example.jobhub.entity.Notification
import com.example.jobhub.model.ApiResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST

interface NotificationService {

    @POST("api/notifications/create-notification")
    fun createNotification(@Body notificationDTO: NotificationDTO): Call<ApiResponse<Void>>

    @GET("api/notifications/get-all-notifications")
    fun getAllNotifications(
        @Header("token") token: String
    ): Call<ApiResponse<List<NotificationEntityDTO>>>

    @POST("api/notifications/mark-as-read")
    fun markAsRead(@Body markAsReadDTO: MarkAsReadDTO): Call<ApiResponse<Void>>
}